src/runner.c

```diff
@@ -0,0 +1,134 @@
+/*
+** © 2018 by Philipp Dunkel, Ben Noordhuis, Elan Shankar
+** Licensed under MIT License.
+*/
+
+#include "runner.h"
+#include <pthread.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <strings.h>
+#include "CoreFoundation/CoreFoundation.h"
+#define NAPI_EXPERIMENTAL
+#include <node_api.h>
+
+typedef struct fse_event {
+  char path[PATH_MAX];
+  size_t pathlen;
+  UInt64 id;
+  UInt32 flags;
+  struct fse_event *next;
+} fse_event_t;
+
+void handleEvents(
+  ConstFSEventStreamRef stream,
+  void *data,
+  size_t numEvents,
+  void *eventPaths,
+  const FSEventStreamEventFlags eventFlags[],
+  const FSEventStreamEventId eventIds[]
+) {
+  size_t idx;
+  fse_t *instance = (fse_t *)data;
+
+  uv_mutex_lock(&instance->mutex);
+  for (idx=0; idx < numEvents; idx++) {
+    fse_event_t *event = (fse_event_t*)malloc(sizeof(fse_event_t));
+    CFStringRef path = (CFStringRef)CFArrayGetValueAtIndex((CFArrayRef)eventPaths, idx);
+    event->pathlen = CFStringGetLength(path);
```

Comment:
Is this correct? `CFStringGetLength()` returns the number of characters, not bytes.

Reply:
No and it's also superfluous